playbooks/installer/ospd/undercloud/pre.yml

```diff
@@ -72,3 +72,8 @@
       lineinfile:
           dest: "/etc/sudoers"
           line: "{{ installer.user.name }} ALL=(root) NOPASSWD:ALL"
+
+- name: Deploy and configure external loadbalancer instance
+  hosts: loadbalancer
+  roles:
+    - { role: "ospd/overcloud/loadbalancer/provision/", when: provisioner.topology.nodes.loadbalancer is defined }
```

Comment:
Also, please have it as it's own role "loadbalancer" and remove the tasks that are osp-d specific to a different role
so basically:
roles/loadbalancer - will install / configure everything needed on the dedicated VM to create the loadbalancer working
roles/installer/ospd/loadbalancer - will configure the loadbalancer according to the OSP-d specific needs / vars.
always keep in mind that future work might not be OSP-d related but still we will want to reuse what you wrote for the LB, so break anything that is not OSP-d related to it's own role